src/build/npm/jobs/npm-package.job.yaml

```diff
@@ -1,17 +1,63 @@
+# Job to build an NPM Package
 parameters:
-  path: ''
-  test: true
-  displayName: 'Build NPM'
-  skipDependencyChecks: false
+  - name: displayName
+    default: 'Build NPM'
+  - name: path
+    default: ''
+  - name: test
+    default: true
+  - name: skipDependencyChecks
+    default: false
 
 jobs:
+  - job:
+    displayName: ${{ parameters.displayName }}
+    steps:
+      - task: npmAuthenticate@0
+        displayName: NPM Authenticate
+        inputs:
+          workingFile: $(Build.SourcesDirectory)/${{parameters.path}}/.npmrc
 
-- job:
-  displayName: ${{ parameters.displayName }}
-  steps:
+      - task: Npm@1
+        displayName: NPM Install
+        inputs:
+          command: custom
+          workingDir: '$(Build.SourcesDirectory)/${{parameters.path}}'
+          customCommand: 'ci --prefer-offline --no-audit'
 
-  - template: /src/build/npm/steps/npm-package.steps.yaml
-    parameters:
-      path: ${{ parameters.path }}
-      test: ${{ parameters.test }}
-      skipDependencyChecks: ${{ parameters.skipDependencyChecks }}
\ No newline at end of file
+      - template: /src/security/dependency-check/tasks/npm-dependency-check.yaml
```

Comment:
these file paths look wrong as there are no tasks in the npm build directory

Reply:
Updated.